modin/distributed/dataframe/pandas/partitions.py

```diff
@@ -78,6 +124,13 @@ def unwrap_partitions(
         )
 
     modin_frame = api_layer_object._query_compiler._modin_frame
+
+    is_modin_distributed = hasattr(modin_frame, "_partition_mgr_cls")
+
+    if not is_modin_distributed and axis == 0:
+        # Directly uses the iloc API to extract data.
+        return _ray_from_modin_non_partition(api_layer_object=api_layer_object)
```

Comment:
this feels wrong - what if user selected non-Ray engine?

Reply:
This is to support ray.data.from_modin, like the code said, it's a hack

Reply:
Can you clarify what scenario this would happen in? Adding a short code example to your linked issue would be helpful.

Unless you're assuming that a user would want to call `ray.data.from_modin` when modin isn't using the Ray backend, you should add an extra if statement to check the [`modin.config.envvars.Engine`](https://github.com/modin-project/modin/blob/01278aeacd9e8943f00a59b2c8715a56d7ba033b/modin/config/envvars.py#L76) environment variable.

Reply:
oh, this thread was about me being stupid; disregard the comment, I misunderstood what code was doing, sorry about that 🙃